File: core/src/commonMain/kotlin/work/socialhub/kslack/api/status/model/SlackIssue.java
package com.github.seratch.jslack.api.status.v2.model;

import java.util.List;

public class SlackIssue {
    private Integer id;
    private String dateCreated;
    private String dateUpdated;
    private String title;
    private String type;
    private String status;
    private String url;
    private List<String> services;
    private List<Note> notes;

    public Integer getId() {
        return this.id;
    }

    public String getDateCreated() {
        return this.dateCreated;
    }

    public String getDateUpdated() {
        return this.dateUpdated;
    }

    public String getTitle() {
        return this.title;
    }

    public String getType() {
        return this.type;
    }

    public String getStatus() {
        return this.status;
    }

    public String getUrl() {
        return this.url;
    }

    public List<String> getServices() {
        return this.services;
    }

    public List<Note> getNotes() {
        return this.notes;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public void setDateCreated(String dateCreated) {
        this.dateCreated = dateCreated;
    }

    public void setDateUpdated(String dateUpdated) {
        this.dateUpdated = dateUpdated;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public void setType(String type) {
        this.type = type;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public void setServices(List<String> services) {
        this.services = services;
    }

    public void setNotes(List<Note> notes) {
        this.notes = notes;
    }

    public static class Note {
        private String dateCreated;
        private String body;

        public String getDateCreated() {
            return this.dateCreated;
        }

        public String getBody() {
            return this.body;
        }

        public void setDateCreated(String dateCreated) {
            this.dateCreated = dateCreated;
        }

        public void setBody(String body) {
            this.body = body;
        }
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/api/status/model/SlackIssue.kt
package work.socialhub.kslack.api.status.model

import kotlinx.serialization.Serializable
import kotlin.js.JsExport

/**
 * Represents a Slack service incident or status update.
 *
 * Contains the incident ID, title, type, current status, affected services,
 * timestamp information, and a timeline of status notes.
 * Queried from the Slack Status API via `status.history()`.
 */
@JsExport
@Serializable
class SlackIssue {
    var id: Int? = null
    var dateCreated: String? = null
    var dateUpdated: String? = null
    var title: String? = null
    var type: String? = null
    var status: String? = null
    var url: String? = null
    var services: Array<String>? = null
    var notes: Array<Note>? = null
}

/**
 * Represents a single update note in a Slack incident timeline.
 *
 * Contains the timestamp and text body of a status change note.
 */
@JsExport
@Serializable
class Note {
    var dateCreated: String? = null
    var body: String? = null
}


